### DISCUSSION TITLE
How to fill field with value from FileUpload component

### DISCUSSION THREAD

**Discussion by Ivanov-Vano** [filamentphp/filament#5974] [Help] [answered]
Hello,

How can I fill the text field with the name of the selected filename using a component FileUpload? I use the "afterStateUpdated" method but it substitutes a different name.

`public static function form(Form $form): Form
    {
        return $form
            ->schema([
                TextInput::make('name')
                    ->required(),
                FileUpload::make('file_path')
                    ->directory('attachments')
                    ->preserveFilenames()
                    ->reactive()
                    ->afterStateUpdated(fn ($state, callable $set) => $set('name', $state)),
            ]);
    }`

With this example of the code, after selecting a file, the value is stored in the text field like as 
"livewire-file:CVp6snZtJOEpQj4EBHVHFlrsHuHKD8-metaZ2l0X2xhYg==-.txt"
but necessary:
"example_filename.txt"

**Comment by danharrin** [comment]
I think you need `storeFileNamesIn('name')` instead of `reactive()->afterStateUpdated()`. `$state` is the file upload object, not a file name, hence why it is being cast as `livewire-file:`

**Reply by Ivanov-Vano** [reply]
Thanks for the fast answer, but I need  that after selecting a file, the field "name" is filled with the filename without extension at once 


![filename](https://user-images.githubusercontent.com/93087459/226209016-a26910f6-0b0e-41e4-8a46-bfc256da3282.png)

**Reply by danharrin** [reply]
My solution will work if you don't have a field

**Reply by ExpertMediaDesign** [reply, marked as answer]
I needed a solution for this as well, and realized Laravel provides a method that works with Filepond's upload object: getClientOriginalName(). More from [the Laravel docs here](https://laravel.com/docs/4.2/requests#files).

With your setup, you can call this on the $state directly:
`->afterStateUpdated(fn ($state, callable $set) => $set('name', $state->getClientOriginalName()))`

**Reply by Ivanov-Vano** [reply]
Thanks a lot! It's work!
